src/frontend/src/session.rs

```diff
@@ -1070,6 +1115,17 @@ impl Session for SessionImpl {
             // TODO: failed transaction
         }
     }
+
+    /// Init and return an `Arc<ExecContext>` which could be used as a guard to represent the execution flow.
+    /// Once
+    fn init_exec_context(&self, sql: Arc<str>) -> ExecContextGuard {
```

Comment:
Does this have to be a method on the `Session` trait? From my perspective, this is an extra functionality that RisingWave's `SessionImpl` provides. `SessionImpl` may maintain it on `handle_query` or `execute` itself, and the trait may not have to be aware of this.

Reply:
Previously, I did put this logic inside `handle_query`, however, I encountered an issue.  For example, a session run `select pg_sleep(10)` should be displayed on the `show processlist`, but `handle_query` itself is not equal to the SQL execution lifecycle(it doesn't `next` the result stream). If we implement `ExecContextGuard` at that level we would consider SQL is ended wrongly. So I have to put this method high level enough (same as the query log).

Reply:
Wow, this is a good catch and makes sense to me. Then the slow query log in `SessionImpl` also appears to be malfunctioning.

https://github.com/risingwavelabs/risingwave/blob/11b2834f555010029b8940db6a5689b19e3adb2d/src/frontend/src/session.rs#L1001-L1014